x-pack/plugins/observability/server/services/slo/reset_slo.test.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+import { elasticsearchServiceMock, loggingSystemMock } from '@kbn/core/server/mocks';
+import { MockedLogger } from '@kbn/logging-mocks';
+
+import { SLO_MODEL_VERSION } from '../../../common/slo/constants';
+import { createSLO } from './fixtures/slo';
+import {
+  createSLORepositoryMock,
+  createSummaryTransformManagerMock,
+  createTransformManagerMock,
+} from './mocks';
+import { ResetSLO } from './reset_slo';
+import { SLORepository } from './slo_repository';
+import { TransformManager } from './transform_manager';
+
+const TEST_DATE = new Date('2023-01-01T00:00:00.000Z');
+
+describe('ResetSLO', () => {
+  let mockRepository: jest.Mocked<SLORepository>;
+  let mockTransformManager: jest.Mocked<TransformManager>;
+  let mockSummaryTransformManager: jest.Mocked<TransformManager>;
+  let mockEsClient: jest.Mocked<ElasticsearchClient>;
+  let loggerMock: jest.Mocked<MockedLogger>;
+  let resetSLO: ResetSLO;
+
+  beforeEach(() => {
+    loggerMock = loggingSystemMock.createLogger();
+    mockRepository = createSLORepositoryMock();
+    mockTransformManager = createTransformManagerMock();
+    mockEsClient = elasticsearchServiceMock.createElasticsearchClient();
+    mockSummaryTransformManager = createSummaryTransformManagerMock();
+    resetSLO = new ResetSLO(
+      mockEsClient,
+      mockRepository,
+      mockTransformManager,
+      mockSummaryTransformManager,
+      loggerMock,
+      'some-space'
+    );
+    jest.useFakeTimers().setSystemTime(TEST_DATE);
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  it('resets all associated resources', async () => {
+    const slo = createSLO({ id: 'irrelevant', version: 1 });
+    mockRepository.findById.mockResolvedValueOnce(slo);
+    mockRepository.save.mockImplementation((v) => Promise.resolve(v));
+
+    await resetSLO.execute(slo.id);
+
+    // delete existing resources and data
+    expect(mockSummaryTransformManager.stop).toMatchSnapshot();
```

Comment:
Why make use of snapshots here? Is there an alternative? I'm not as familiar with using snapshots in this way (against mock functions).

Reply:
Discussed this in another comment.